### ISSUE TITLE
Gaia Crusaders Audio Pitched Higher than Normal

### ISSUE THREAD

**Issue by Awakened0** [libretro/mame#18]
This game (gaia) and probably some others have their audio pitched higher in the libretro nightly 2-9-2015 core than in stand alone MAME 0.158. I think it's because the game is running a little faster than normal in the libretro version. I tested the 1.0.0.3 beta build of the MAME core and that version sounds the same as stand alone 0.158, so the change that causes it must have happened after 10-29-2014.

Here's a comparison I recorded: https://www.dropbox.com/s/2sbz32ppa5znefl/gaia.mp3?dl=0

The first half is stand alone and the second that starts at about 0:24 is libretro.

**Comment by Awakened0** [comment]
Looks like it's refresh rate related, since I can set it to 57 in my config and get it pretty close to stand alone. I wonder if there's a better way though.

**Comment by Awakened0** [comment]
I tried out the "throttle off, wait for vsync and refresh speed on" stand alone settings mentioned in [this thread](http://libretro.com/forums/threads/2016-MAME-2015-Refresh-Rate-Audio-Video-Sync-Questions?p=18548#post18548) and managed to get perfectly synced visuals (no flickering shadows) without any sound distortion like what I hear in the libretro core.

**Comment by Awakened0** [comment]
I discovered changing "audio_max_timing_skew" from 0.05 to 0.00 in the nightly will fix the audio, but cause flickering shadows.

**Comment by hizzlekizzle** [comment]
I think audio_max_timing_skew is intended for this case, exactly. You're actually supposed to turn it up until there's no more crackling and the refresh-locked effects should function properly at your monitor's refresh and the audio will be higher-pitched, as the whole game is running slightly faster than intended.

I'm not really sure what standalone could be doing to make the game run faster visually but without either speeding up the audio or getting a pop every time the audio buffer empties, unless it actually is running the audio faster and is just resampling it at a lower pitch to hide it.

**Comment by Awakened0** [comment]
Could be, though I'm not sure how I'd verify that. In my stand alone mame.ini I have my sample rate set to 48000, just like RetroArch. There's also an audio latency setting set to 2. I tried setting it to 0 and it didn't affect the pitch.

I did a recording test in stand alone: https://www.dropbox.com/s/u1gf0jq8da3agb9/gaia2.mp3?dl=0

First half is with throttle on, wait for vsync and refresh speed off. Second half (starts at about 0.26) is throttle off and the other two on. Pitch wise they both sound the same, though there are some audio pops in the second. I tried a few settings combinations with sync refresh and refresh speed on and off and they didn't affect the pitch either.
